aws/credentials/stscreds/assume_role_provider.go

```diff
@@ -0,0 +1,116 @@
+package stscreds
+
+import (
+	"fmt"
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/credentials"
+	"github.com/aws/aws-sdk-go/service/sts"
+	"time"
+)
+
+// assumeRoler represents the minimal subset of the STS API used here.
+type assumeRoler interface {
```

Comment:
@jasdel I can't really stub the STS client because it's an autogenerated struct. Do you think I could use a new private `assumeRoler` interface like that?